src/dotnet-ildasm/Infrastructure/TypeReferenceExtensions.cs

```diff
@@ -5,22 +5,27 @@ namespace DotNet.Ildasm.Infrastructure
 {
     public static class TypeReferenceExtensions
     {
-        public static string ToILType(this TypeReference typeReference)
+        public static string ToILNameFormatt(this TypeReference typeReference)
         {
-            if (typeReference.MetadataType == MetadataType.Void)
-                return "void";
-            if (typeReference.MetadataType == MetadataType.String)
-                return "string";
-            if (typeReference.MetadataType == MetadataType.Int32)
-                return "int";
-            if (typeReference.MetadataType == MetadataType.Array)
-                return $"{typeReference.GetElementType().ToILType()}[]";
-
             if (string.Compare(typeReference.Scope.Name, typeReference.Module.Name,
                     StringComparison.CurrentCultureIgnoreCase) == 0)
                 return $"{typeReference.FullName}";
 
             return $"[{typeReference.Scope.Name}]{typeReference.FullName}";
         }
+
+        public static string ToILType(this TypeReference typeReference)
+        {
+            if(typeReference.MetadataType == MetadataType.ValueType || typeReference.MetadataType == MetadataType.Class)
+                return $"{typeReference.MetadataType.ToString().ToLowerInvariant()} {ToILNameFormatt(typeReference)}";
+
+            if (typeReference.MetadataType != MetadataType.Array)
+                return typeReference.MetadataType.ToString().ToLowerInvariant();
+
+            if (typeReference.MetadataType == MetadataType.Array)
+                return $"{typeReference.GetElementType().ToILType()}[]";
+
+            throw new NotSupportedException();
```

Comment:
What exactly scenarios are you covering here? Can you please create tests to cover them?